### PR TITLE
[MRG] Fix constant prediction in bayesian optimisation

### PR DESCRIPTION
**Pull request by MechCoder**
Fixes https://github.com/scikit-optimize/scikit-optimize/issues/252 partially

### CODE REVIEW

**Comment by MechCoder** [comment]
Should we stop promising the optimisation of noisy functions until we are clear about what we are doing?

**Comment by glouppe** [comment]
Maybe we should also have more than 10 iterations, since the last plot is not very convincing at the moment (it is far from having converged).

**Comment by MechCoder** [comment]
I tried that but it gives a flat prediction after that.

File: skopt/optimizer/gp.py [hunk 1]
```diff
@@ -149,6 +150,14 @@ def gp_minimize(func, dimensions, base_estimator=None,
         Controls how much improvement one wants over the previous best
         values. Used when the acquisition is either `"EI"` or `"PI"`.
 
+    * `noise` [float, default="gaussian"]:
+        - If `noise="gaussian"` the function is noisy and returns
+          being `f(x_i) + e(i)` where e(i) is zero centered and has a fixed
+          variance.
+        - If the variance is known before-hand, this can be set directly
+          to the variance of the noise.
+        - Set this to zero if the function is non-noisy.
```
Comment (betatim) [inline skopt/optimizer/gp.py]: Set this to zero is the objective is noise free? (non-noisy seems odd)

**Comment by betatim** [comment]
(Nitpick: could you remove `plt.rcParams["figure.figsize"] = (10, 6)` at the top of the notebook so we get smaller figures?)

**Comment by MechCoder** [comment]
I'll do making the plots smaller just before the release for all examples together. I've addressed everything else.

**Comment by betatim** [comment]
Thanks a lot!

**Comment by glouppe** [comment]
:beers:

**Pull request merged**